Tasks/PublishSymbols/Invoke-IndexSources.ps1

```diff
@@ -35,6 +35,63 @@ if (!(Get-Command @splat)) {
         [System.Reflection.BindingFlags]::Instance -bor [System.Reflection.BindingFlags]::NonPublic) # bindingFlags
 }
 
+function Add-DbghelpLibrary {
+    [cmdletbinding()]
+    param()
+
+    # Check if dbghelp.dll is already loaded.
+    Write-Verbose "Checking if library is already loaded: dbghelp.dll"
+    [string]$filePath = "$env:AGENT_HOMEDIRECTORY\Agent\Worker\Tools\Symstore\dbghelp.dll"
+    [bool]$isLoaded = $false
+    [System.Diagnostics.Process]$process = [System.Diagnostics.Process]::GetCurrentProcess();
+    foreach ($module in $process.Modules) {
+        if ($module.ModuleName -eq 'dbghelp.dll') {
+            $isLoaded = $true
+            if ($module.FileName -eq $filePath) {
+                Write-Verbose "Module dbghelp.dll is already loaded from the expected file path: $filePath"
+            } else {
+                Write-Warning (Get-LocalizedString -Key "Library dbghelp.dll is already loaded from an unexpected file path: {0} ; Expected path: {1} ; An incorrect version of the library may result in malformed source file paths to be extracted from the PDB files. If this condition occurs, it will be indicated in the logs below." -ArgumentList $($module.FileName), $filePath)
+            }
+
+            # Don't short-circuit the loop. The module could be loaded more
+            # than once and we should trace info about each loaded instance.
+        }
+    }
+
+    # Short-ciruit if already loaded.
+    if ($isLoaded) {
```

Comment:
I def have not much clue on the stuff here, but can't we return inside foreach? it does seem like we would loop further though we reached a point where we have to return..